patches/minecraft/net/minecraft/entity/passive/HorseArmorType.java.patch

```diff
@@ -0,0 +1,64 @@
+--- ../src-base/minecraft/net/minecraft/entity/passive/HorseArmorType.java
++++ ../src-work/minecraft/net/minecraft/entity/passive/HorseArmorType.java
+@@ -37,7 +37,6 @@
+         return this.ordinal();
+     }
+ 
+-    @SideOnly(Side.CLIENT)
+     public String func_188573_b()
+     {
+         return this.field_188587_f;
+@@ -55,6 +54,7 @@
+         return this.field_188586_e;
+     }
+ 
++    @Deprecated /**Forge: Use getByName. Ordinals of mod-added enum constants are dependent on load order, unlike names.**/
+     public static HorseArmorType func_188575_a(int p_188575_0_)
+     {
+         return values()[p_188575_0_];
+@@ -62,9 +62,14 @@
+ 
+     public static HorseArmorType func_188580_a(ItemStack p_188580_0_)
+     {
+-        return p_188580_0_.func_190926_b() ? NONE : func_188576_a(p_188580_0_.func_77973_b());
++        if (p_188580_0_.func_77973_b() instanceof net.minecraftforge.common.IHorseArmor)
++        {
++            return ((net.minecraftforge.common.IHorseArmor) p_188580_0_.func_77973_b()).getHorseArmorType(p_188580_0_);
++        }
++        return p_188580_0_.func_77973_b() == Items.field_151138_bX ? net.minecraft.entity.passive.HorseArmorType.IRON : p_188580_0_.func_77973_b() == Items.field_151136_bY ? net.minecraft.entity.passive.HorseArmorType.GOLD : p_188580_0_.func_77973_b() == Items.field_151125_bZ ? net.minecraft.entity.passive.HorseArmorType.DIAMOND : net.minecraft.entity.passive.HorseArmorType.NONE;
+     }
+ 
++    @Deprecated //Forge: Use getByItemStack
+     public static HorseArmorType func_188576_a(Item p_188576_0_)
+     {
+         if (p_188576_0_ == Items.field_151138_bX)
+@@ -81,8 +86,29 @@
+         }
+     }
+ 
++    @Deprecated //Forge: Use ItemStack sensitive overload
+     public static boolean func_188577_b(Item p_188577_0_)
+     {
+         return func_188576_a(p_188577_0_) != NONE;
+     }
++    
++    /********* FORGE *********/
```

Comment:
Dangling javadocs

Reply:
That's what mezz told me to put there. There's no standard for forge section delimiters, so I'm sticking with that.

Reply:
```java
/* ======================================== FORGE START ===================================== */
```
seems pretty standard.

If you want to stick with that, please do use spaces inside the `/*` and `*/` parts to delimit the comment markers from the actual comment. As you have it now, it's an erroneous Javadoc comment.

Reply:
Valid point

Reply:
I usually just hold down = until it fills up 1/2 my screen :P it's intended to be fairly big and obvious.